polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1437 @@
+                cell_b (dict) - End cell environmental info as dictionary
+                x (tuple) - Updated crossing point that could lie outside the connection of the cell boxes (long,lat)
+                case (int) - Adjaceny case tupe connecting the two cells
+
+            Output
+                case_a (int or None) - Possible additional case edge relative to start cell to add, if None no edge to add
+                case_b (int or None) - Possible additional case edge relative to start cell to add, if None no edge to add
+            
+        '''   
+
+        # Returning if corner horseshoe case type
+        if abs(case)==1 or abs(case)==3 or abs(case)==0:
+            return None,None
+        elif abs(case) == 2:
+            # Defining the min and max of the start and end cells
+            smin = cell_a['cy']-cell_a['dcy'] 
+            smax = cell_a['cy']+cell_a['dcy']
+            emin = cell_b['cy']-cell_b['dcy']
+            emax = cell_b['cy']+cell_b['dcy']
+
+            # Defining the global min and max
+            vmin = np.max([smin,emin])
+            vmax = np.min([smax,emax])
+
+            # Point crossingpoint on boundary between the two origional cells
+            if (x[1] >= vmin) and (x[1] <= vmax):
+                return None,None
+
+            # If Start and end cells share a edge for the horesshoe 
+            if (x[1]<smin) and (smin==emin):
+                case_a = 4
+                case_b   = 4
+            if (x[1]>smax) and (smax==emax):
+                case_a = -4
+                case_b   = -4
+
+            # --- Cases where StartCell is Larger than end Cell ---
+            if (x[1]>emax) and (smax>emax):
+                case_a = case
+                case_b   = (-4)                
+            if (x[1]<emin) and (smin<emin):
+                case_a = case
+                case_b   = (4)                   
+
+            # --- Cases where StartCell is smaller than end Cell ---
+            if (x[1]>smax) and (smax<emax):
+                case_a = -4
+                case_b   = -case
+            if (x[1]<smin) and (emin<smin):
+                case_a = 4
+                case_b   = -case      
+
+        elif abs(case) == 4:
+            # Defining the min and max of the start and end cells
+            smin = cell_a['cx']-cell_a['dcx']
+            smax = cell_a['cx']+cell_a['dcx']
+            emin = cell_b['cx']-cell_b['dcx']
+            emax = cell_b['cx']+cell_b['dcx']
+
+            # Defining the global min and max
+            vmin = np.max([smin,emin])
+            vmax = np.min([smax,emax])
+
+            # Point crossingpoint on boundary between the two origional cells
+            if (x[0] >= vmin) and (x[0] <= vmax):
+                return None,None
+
+            # If Start and end cells share a edge for the horesshoe 
+            if (x[0]<smin) and (smin==emin):
+                case_a = -2
+                case_b   = -2
```

Comment:
```
if (x[0]<smin) and (smin==emin):
        return -2, -2
```
Would this work?

Reply:
This would work, but I think it's better to spell out the case additions of case_a,case_b. 

I am happy to change if needed